.github/workflows/model_image_build_push.yaml

```diff
@@ -61,6 +67,108 @@ jobs:
           image: ${{ steps.build_convert_models_image.outputs.image }}
           tags: ${{ steps.build_convert_models_image.outputs.tags }}
 
+  build-and-push-granite-model-image:
+    if: contains( github.event.pull_request.labels.*.name, 'hold-tests')
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      # need enough disk space to download model
+      - name: Remove unnecessary files
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+
+      - uses: actions/checkout@v4.1.1
+
+      # required for multi-arch builds
+      - name: Install qemu dependency
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y qemu-user-static
+
+      - name: Build granite model image
+        id: build_granite_model_image
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: ${{ env.GRANITE_MODEL_IMAGE_NAME }}
```

Comment:
Based on my above suggestion this will move to a matrix based approach, ex [ms model downloads via matrix]((https://github.com/containers/ai-lab-recipes/blob/main/.github/workflows/model_servers.yaml#L79-L81)